components/shell/bash-completion/patches/02-gawk.patch

```diff
@@ -0,0 +1,1254 @@
++            $( gawk '! /\*/ { print $NF }' /etc/filesystems 2>/dev/null )"
+     else
+         # Generic
+-        fss="$( awk '/^[ \t]*[^#]/ { print $3 }' /etc/fstab 2>/dev/null )
+-            $( awk '/^[ \t]*[^#]/ { print $3 }' /etc/mnttab 2>/dev/null )
+-            $( awk '/^[ \t]*[^#]/ { print $4 }' /etc/vfstab 2>/dev/null )
+-            $( awk '{ print $1 }' /etc/dfs/fstypes 2>/dev/null )
++        fss="$( gawk '/^[ \t]*[^#]/ { print $3 }' /etc/fstab 2>/dev/null )
++            $( gawk '/^[ \t]*[^#]/ { print $3 }' /etc/mnttab 2>/dev/null )
++            $( gawk '/^[ \t]*[^#]/ { print $4 }' /etc/vfstab 2>/dev/null )
++            $( gawk '{ print $1 }' /etc/dfs/fstypes 2>/dev/null )
+             $( [[ -d /etc/fs ]] && command ls /etc/fs )"
+     fi
+ 
+@@ -1364,7 +1364,7 @@ _count_args()
+ _pci_ids()
+ {
+     COMPREPLY+=( $( compgen -W \
+-        "$( PATH="$PATH:/sbin" lspci -n | awk '{print $3}')" -- "$cur" ) )
++        "$( PATH="$PATH:/sbin" lspci -n | gawk '{print $3}')" -- "$cur" ) )
+ }
+ 
+ # This function completes on USB IDs
+@@ -1372,7 +1372,7 @@ _pci_ids()
+ _usb_ids()
+ {
+     COMPREPLY+=( $( compgen -W \
+-        "$( PATH="$PATH:/sbin" lsusb | awk '{print $6}' )" -- "$cur" ) )
++        "$( PATH="$PATH:/sbin" lsusb | gawk '{print $6}' )" -- "$cur" ) )
+ }
+ 
+ # CD device names
+@@ -1394,7 +1394,7 @@ _terms()
+         "$( command sed -ne 's/^\([^[:space:]#|]\{2,\}\)|.*/\1/p' /etc/termcap \
+             2>/dev/null )" -- "$cur" ) )
+     COMPREPLY+=( $( compgen -W "$( { toe -a 2>/dev/null || toe 2>/dev/null; } \
+-        | awk '{ print $1 }' | sort -u )" -- "$cur" ) )
++        | gawk '{ print $1 }' | sort -u )" -- "$cur" ) )
+ }
+ 
+ # a little help for FreeBSD ports users
+@@ -1483,7 +1483,7 @@ _included_ssh_config_files()
+ _known_hosts_real()
+ {
+     local configfile flag prefix
+-    local cur curd awkcur user suffix aliases i host ipv4 ipv6
++    local cur curd gawkcur user suffix aliases i host ipv4 ipv6
+     local -a kh khd config
+ 
+     # TODO remove trailing %foo from entries
+@@ -1529,7 +1529,7 @@ _known_hosts_real()
+         # TODO(?): try to make known hosts files with more than one consecutive
+         #          spaces in their name work (watch out for ~ expansion
+         #          breakage! Alioth#311595)
+-        tmpkh=( $( awk 'sub("^[ \t]*([Gg][Ll][Oo][Bb][Aa][Ll]|[Uu][Ss][Ee][Rr])[Kk][Nn][Oo][Ww][Nn][Hh][Oo][Ss][Tt][Ss][Ff][Ii][Ll][Ee][ \t]+", "") { print $0 }' "${config[@]}" | sort -u ) )
++        tmpkh=( $( gawk 'sub("^[ \t]*([Gg][Ll][Oo][Bb][Aa][Ll]|[Uu][Ss][Ee][Rr])[Kk][Nn][Oo][Ww][Nn][Hh][Oo][Ss][Tt][Ss][Ff][Ii][Ll][Ee][ \t]+", "") { print $0 }' "${config[@]}" | sort -u ) )
+         IFS=$OIFS
+         for i in "${tmpkh[@]}"; do
+             # First deal with quoted entries...
+@@ -1561,34 +1561,34 @@ _known_hosts_real()
+ 
+     # If we have known_hosts files to use
+     if [[ ${#kh[@]} -gt 0 || ${#khd[@]} -gt 0 ]]; then
+-        # Escape slashes and dots in paths for awk
```

Comment:
Generally, patch touches too much. Please, review it and remove unnecessary changes.